aws-datastore/src/main/java/com/amplifyframework/datastore/appsync/AppSyncRequestFactory.java

```diff
@@ -67,7 +69,22 @@
  * and AppSync-specific field names (`_version`, `_deleted`, etc.)
  */
 final class AppSyncRequestFactory {
-    private AppSyncRequestFactory() {}
+    private final ModelSchemaRegistry modelSchemaRegistry;
+
+    /**
+     * Constructs an AppSyncRequestFactory using an empty schema registry.
+     */
+    AppSyncRequestFactory() {
+        this.modelSchemaRegistry = ModelSchemaRegistry.instance();
```

Comment:
What's the purpose of this constructor?  Shouldn't a `ModelSchemaRegistry` always be passed in, instead of just creating an empty one?

Reply:
I started out requiring it to be passed. I added this default constructor only to reduce the amount of changes needed to snake the repository around. But, I agree with your concern - this would be hard to track down later, and could cause sneaky bugs. I'll remove this version of the constructor.